funsor/sum_product.py

```diff
@@ -181,7 +181,10 @@ def sequential_sum_product(sum_op, prod_op, trans, time, step):
     return trans(**{time: 0})
 
 
-def naive_sarkka_bilmes_product(sum_op, prod_op, trans, time_var):
+def naive_sarkka_bilmes_product(sum_op, prod_op, trans, time_var, global_vars=None):
```

Comment:
nit: may as well default to `global_vars=frozenset()`. (in general it is dangerous to make defaults mutable data structures, but (deeply) immutable data structures are safe)